pkg/provisioner/controller.go

```diff
@@ -74,15 +75,27 @@ func NewController(provisionerName string, provisioner api.Provisioner, clientse
 
 	obcInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: ctrl.enqueueOBC,
-		UpdateFunc: func(oldObj, newObj interface{}) {
-			// noop
+		UpdateFunc: func(old, new interface{}) {
+			oldObc := old.(*v1alpha1.ObjectBucketClaim)
+			newObc := new.(*v1alpha1.ObjectBucketClaim)
+			if newObc.ResourceVersion == oldObc.ResourceVersion {
+				// periodic re-sync can be ignored
+				return
+			}
+			// if old and new both have deletionTimestamps we can also ignore the
+			// update since these events are occurring on an obc marked for deletion,
+			// eg. extra finalizers being added and deleted.
+			if newObc.ObjectMeta.DeletionTimestamp != nil && oldObc.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
Perhaps. If this suggestion is agreed to then I think it should be a separate pr.
There is always the manual step of deleting the finalizer on a stuck resource, but I appreciate the desire to automate this to some degree.

Reply:
Separate PR? Why? I just want to remove these 3 lines checking for:
```
if newObc.ObjectMeta.DeletionTimestamp != nil && oldObc.ObjectMeta.DeletionTimestamp != nil {
    return
}
```

Reply:
If those lines are removed every time the obc is updated after deletion (there are ~5 times this occurs in the path of a delete) our reconciliation code will be executed. The code will attempt to re-delete the resources associated with the obc. It may makes more sense to retry deletion of the various resources within a single reconciliation cycle.

Reply:
Sounds like the problem is that we update the deleted obc 5 times. What is the reason for doing that?

Reply:
k8s adds a finalizer to the parent of each dependent resource (via ownerReference). So 2 updates for that. Then k8s removes the finalizers it added once the cm and secret are deleted, so 2 more updates. I thought there an additional update but don't remember now...

Reply:
Adding a finalizer is to owner might be related to using `blockOwnerDeletion`. I can say for sure that an ownerReference without blockOwnerDeletion does not add a finalizer to the owner. Do we really need to set blockOwnerDeletion? Once we remove the finalizer from Secret/CM/OB their deletion is "non-blocking" anyhow so the owner doesn't need to be wait for them so we can get rid of all these finalizer updates. Yes yes, in a separate PR... :)

Reply:
I don't fully know the behavior of blockOwnerDeletion. From the doc and code snippets we examined this seemed like the correct setting. However, maybe it is related to the lib addin finalizers to the secret and cm?
Overall, this pr works (as much as I've been able to test) as is and I think he should get merged. These concerns/comments are good but I feel they should be added to a new issue with a new pr to resolve them.